content/en/docs/tasks/observability/metrics/tcp-metrics/test.sh

```diff
@@ -51,16 +51,28 @@ _wait_for_istio virtualservice default ratings
 
 # Get GATEWAY_URL
 # export the INGRESS_ environment variables
-_set_ingress_environment_variables
+# TODO make this work more generally. Currently using snips for Kind.
+export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+export SECURE_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+export INGRESS_HOST=$(kubectl get po -l istio=ingressgateway -n istio-system -o jsonpath='{.items[0].status.hostIP}')
 export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
-snip_collecting_new_telemetry_data_9
 
-# Next steps talk about looking at the Prometheus browser
-# We dont have any browser testing
-# Not sure if we can curl proetheus (I didn't get to work)
-#   ex. curl -g 'http://127.0.0.1:9090/api/v1/query?query=istio_tcp_connections_closed_total'
-# snip_collecting_new_telemetry_data_10
-# killall kubectl
+# Hit the GATEWAY_URL
+# snip_collecting_new_telemetry_data_9
+get_bookinfo_productpage() {
+    curl -s "http://${GATEWAY_URL}/productpage" | grep -o "<title>.*</title>"
+ }
+ _verify_contains get_bookinfo_productpage "<title>Simple Bookstore App</title>"
+
+# Next steps look at the Prometheus browser
+# We will use istioctl to query some metrics to verify Prometheus
+metrics_output="                                  WORKLOAD    TOTAL RPS    ERROR RPS  P50 LATENCY  P90 LATENCY  P99 LATENCY
+                    productpage-v1.default        0.000        0.000           0s           0s           0s"
```

Comment:
do we want something that shows off TCP metrics instead?  this dashboard is HTTP focused.

Reply:
Well, the page has the section: `Verify that the TCP metric values are being generated and collected.` and has them port forward to prometheus, and then in the dashboard, have then have them get the istio_tcp_connections_opened_total metric and has a sample console output for said.

What can we do that would show that it's available. I did try and run a curl command to get the output, but wasn't totally successful.

At least I'm verifying that some metrics are available.

Reply:
I think this really should try to confirm that the metrics that the doc is referring too are there, if possible.

Reply:
@frankbu I did add steps to verify that the 1st metric that we have the reader enter in the dashboard, istio_tcp_connections_opened_total, does get returned via a curl to prometheus.